chainspec_test.go

```diff
@@ -0,0 +1,166 @@
+package ibctest_test
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/strangelove-ventures/ibctest"
+	"github.com/strangelove-ventures/ibctest/ibc"
+	"github.com/stretchr/testify/require"
+)
+
+func TestChainSpec_Config(t *testing.T) {
+	t.Run("valid", func(t *testing.T) {
+		s := ibctest.ChainSpec{
+			Name: "gaia",
+
+			Version: "v7.0.1",
+		}
+
+		_, err := s.Config()
+		require.NoError(t, err)
+	})
+
+	t.Run("omit name when all other fields provided", func(t *testing.T) {
+		s := ibctest.ChainSpec{
+			ChainName: "mychain",
+
+			ChainConfig: ibc.ChainConfig{
+				Type: "cosmos",
+				// Skip Name, as that is intended to be inherited from ChainName.
+				ChainID: "mychain-123",
+				Images: []ibc.DockerImage{
+					{Repository: "docker.example.com", Version: "latest"},
+				},
```

Comment:
I know this was part of the original design, but I'm surprised `Images` is an array vs. one image. However, it's used for penumbra which also needs a tendermint image. 

No need for changes, just a general comment. 

My first instinct is you can't configure highly custom, one-offs like penumbra. It must all be implemented in code, sidestepping this config, which may help simplify the config here. Granted, there may be a common pattern with tendermint as a "sidecar" as seen with penumbra. 

Also (from the previous pattern) comma delimited docker image version is not clear.